lib/diagnostic.js

```diff
@@ -5,22 +5,36 @@
 // and returns the minimum from that set
 // assume that all the argument are numbers
 const min = function min() {
-
+  let result = arguments[0];
+  for (let i=1;i<arguments.length;i++){
+    if (result > arguments[i]){
+      result = arguments[i];
+    }
+  }
+  return result
 };
 
 // Question 2
 // Write a function that takes zero or more arguments
 // and returns the sum of that set
 // assume that all the argument are numbers
 const sum = function sum() {
-
+  let result = 0;
+  for (let i=0;i<arguments.length;i++){
+    result += arguments[i]
+  }
+return result
 };
 
 // Question 3
 // Write a function that returns an array of the specified length
 // with all the elements set to defaultValue
 const newArray = function newArray(length, defaultValue) {
-
+  let array = [];
+  for(let i=0;i<=length-1;i++){
+    array[i] = defaultValue(i);
+  }
```

Comment:
Think about what defaultValue(i) means--javascript is interpreting this as a function call 'defaultValue' and passing it whatever i is in the for loop as a parameter.